File: 其他/LeetCode算法/calcu/2020.8 - now/10.16/剑指 Offer 06. 从尾到头打印链表.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {number[]}
 */
var reversePrint = function(head) {
  let ans = []
  while (head.val) {
    ans.splice(0, 0, head.val)
    head = head.next
  }
  return ans
};

File: 其他/LeetCode算法/calcu/2020.8 - 6.15/10.12/剑指 Offer 06. 从尾到头打印链表.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class Solution:
    def reversePrint(self, head: ListNode) -> List[int]:
      # dq = []
      # while head :
      #   dq.insert(0, head.val)
      #   head = head.next
      # return dq

      # 递归
      return self.reversePrint(head.val) + [head.val] if head else []